### ISSUE TITLE
pip install cleanx, on mac errors

### ISSUE THREAD

**Issue by henrykironde** [drcandacemakedamoore/cleanX#36]
**Describe the bug**
  No package 'tesseract' found

**Screenshots**

```python
Using legacy 'setup.py install' for tesserocr, since package 'wheel' is not installed.
Installing collected packages: tesserocr, opencv-python, matplotlib, cleanX
    Running setup.py install for tesserocr ... error
    ERROR: Command errored out with exit status 1:
     command: /Users/henrykironde/Documents/GitHub/testenv/bin/python3.9 -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/d5/vm7jfw2x7q550xrltd8ss5440000gn/T/pip-install-d2ki3n8m/tesserocr_9008e8f2373142109467f5269239caa5/setup.py'"'"'; __file__='"'"'/private/var/folders/d5/vm7jfw2x7q550xrltd8ss5440000gn/T/pip-install-d2ki3n8m/tesserocr_9008e8f2373142109467f5269239caa5/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /private/var/folders/d5/vm7jfw2x7q550xrltd8ss5440000gn/T/pip-record-zx09fpub/install-record.txt --single-version-externally-managed --compile --install-headers /Users/henrykironde/Documents/GitHub/testenv/include/site/python3.9/tesserocr
         cwd: /private/var/folders/d5/vm7jfw2x7q550xrltd8ss5440000gn/T/pip-install-d2ki3n8m/tesserocr_9008e8f2373142109467f5269239caa5/
    Complete output (20 lines):
    pkg-config failed to find tesseract/leptonica libraries: Package tesseract was not found in the pkg-config search path.
    Perhaps you should add the directory containing `tesseract.pc'
    to the PKG_CONFIG_PATH environment variable
    No package 'tesseract' found
    
    Failed to extract tesseract version from executable: [Errno 2] No such file or directory: 'tesseract'
    Supporting tesseract v3.04.00
    Tesseract major version 3
    Building with configs: {'libraries': ['tesseract', 'lept'], 'compile_time_env': {'TESSERACT_MAJOR_VERSION': 3, 'TESSERACT_VERSION': 50593792}}
    WARNING: The wheel package is not available.
    running install
    running build
    running build_ext
    Detected compiler: unix
    building 'tesserocr' extension
    creating build
    creating build/temp.macosx-11-x86_64-3.9
    clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -I/usr/local/include -I/usr/local/opt/openssl@1.1/include -I/usr/local/opt/sqlite/include -I/Users/henry/Documents/GitHub/testenv/include -I/usr/local/opt/python@3.9/Frameworks/Python.framework/Versions/3.9/include/python3.9 -c tesserocr.cpp -o build/temp.macosx-11-x86_64-3.9/tesserocr.o
    clang: error: invalid version number in 'MACOSX_DEPLOYMENT_TARGET=11'
    error: command '/usr/bin/clang' failed with exit code 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /Users/henrykironde/Documents/GitHub/testenv/bin/python3.9 -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/d5/vm7jfw2x7q550xrltd8ss5440000gn/T/pip-install-d2ki3n8m/tesserocr_9008e8f2373142109467f5269239caa5/setup.py'"'"'; __file__='"'"'/private/var/folders/d5/vm7jfw2x7q550xrltd8ss5440000gn/T/pip-install-d2ki3n8m/tesserocr_9008e8f2373142109467f5269239caa5/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /private/var/folders/d5/vm7jfw2x7q550xrltd8ss5440000gn/T/pip-record-zx09fpub/install-record.txt --single-version-externally-managed --compile --install-headers /Users/henrykironde/Documents/GitHub/testenv/include/site/python3.9/tesserocr Check the logs for full command output.
WARNING: You are using pip version 21.1.3; however, version 21.2.4 is available.

(testenv) ➜  cleanX git:(docs) ✗ 
```

**Your computer environment info: (please complete the following information):**

    OS: [MacOSX]
    Python V=version [3.9]

**Comment by drcandacemakedamoore** [comment]
Assigned to Oleg Sivokon.

**Comment by wvxvw** [comment]
We don't actually test `cleanX` with Python from python.org on systems other than Linux. We only have Anaconda tests for them.

Similarly, on MS Windows, you'd have to find a way to install Tesseract shared library.

Neither me nor @drcandacemakedamoore have a Mac to test how it works. Anaconda was relatively simple to support, as it uses the same process for installing shared libraries on Linux and everywhere else. But I would need at least a VM running flavors of OSX and Windows to start to understand how to deal with these dependencies if not through Anaconda.

----

Bottom line, for now, if you want to run it on anything other than Linux, you either need to use Anaconda, or you are on your own installing Tesseract shared libraries.

I'd try this:

``` sh
brew install tesseract
```

And hope for the best. I don't actually know what it will do and whether it will work. On Linux, `tesseract` is a number of different packages, mostly because different languages ares split in their own packages. But, maybe for install to succeed the above will be enough, if it works.

**Comment by drcandacemakedamoore** [comment]
I just noticed Anaconda seems to be on ou3 1.8 version. I will look into it and try to fix it to 1.9 today.

**Comment by henrykironde** [comment]
I would also add the system requirements to the readme file to let people know the easy way to install on the various supported OS

**Comment by wvxvw** [comment]
@henrykironde We've done it already. See starting here: https://github.com/drcandacemakedamoore/cleanX/blame/main/README.md#L47

**Comment by drcandacemakedamoore** [comment]
Just a note on my earlier comment about bringing everything up to 1.9: in some hours today everything will go under the new versions 1.10 or perhaps 1.11, not 1.9 for technical reasons. We are trying to limit releases, but this may be unavoidable. Right now everything but Windows is up to 1.10. Whatever the case we should have everything the new version out in less than 24 hours now. Thank you for your patience reviewer.

**Comment by drcandacemakedamoore** [comment]
In the new version, we have updated the readme file to show what is supported in a fuller way. I am therefore closing this issue.
